consent/helpers.py

```diff
@@ -19,6 +20,18 @@ def consent_exists(username, course_id, enterprise_customer_uuid):
     :param enterprise_customer_uuid: The consent requester.
     :return: Whether any consent (provided or unprovided) exists.
     """
+    # pylint: disable=invalid-name
+    EnterpriseCourseEnrollment = apps.get_model('enterprise', 'EnterpriseCourseEnrollment')
+    if EnterpriseCourseEnrollment.objects.filter(
+            course_id=course_id,
+            enterprise_customer_user__user_id=get_user_id(username),
+            enterprise_customer_user__enterprise_customer__uuid=enterprise_customer_uuid,
+    ).exists():
+        # We're not creating consent records when we do proxy enrollment, so
+        # check for an enrollment as an indicator that consent does actually exist,
```

Comment:
Clarification: We're not looking for whether _consent_ exists; we're looking for whether a situation in which any record related to the overall consent scenario exists.